### ISSUE TITLE
Pencil2D Website is Down

### ISSUE THREAD

**Issue by Jose-Moreno** [pencil2d/pencil#346]
Hello everyone,

I don't know if anyone noticed but the Pencil2D.org website is down. I dont know what has happened as according to google it was working today until 7 PM GMT-5

With fear of all of your effort going to waste, I decided to create a temporary free webpage, plus a forum in case we can't have our old community back. Either way i'll be maintaining that one if one day someone decides to pull the plug once and for all.

For now I'll share the forum since it's easier. I'm still creating the site to make it usable, so for now the forum will do. Give me until Nov 30th so I can share the rest, and also so I can be more active towards looking how we can improve this app and how we can find more devs.

http://pencil2d.forumotion.net/

**Comment by Jose-Moreno** [comment]
This issue is not related to the app itself but rather is a matter of urgency for everyone in the Pencil community. Please label it as question or else, but I didn't know where else I could contact some of you who are active.

**Comment by feeef** [comment]
Thanks for that @Jose-Moreno ! I am finishing my updates on the vector engine and I prepare a screencast to comment my work. I am pretty happy with what I have done so far.
I am still confident we will have a working version in a couple of weeks. I have fixed what was the hardest to fix. It would be a shame throwing it away right now. If Matt has decided to stop the project and the website we should ask him to send an email to the community to tell them we want to keep working on Pencil and point them to a new website.

**Comment by feeef** [comment]
The website is back! ;) I have made my screencast video and published it here : https://goblinrefuge.com/mediagoblin/u/feeef/m/pencil-dev-update-colours-on-vector-engine/
The quality of the inline video may not be very good but you can check the original file (there is a download link on the page) if you want better quality

**Comment by chchwy** [comment]
Pencil2D.org is back, I'm going to close it.

**Comment by Jose-Moreno** [comment]
@chchwy @gordielachance 22/01/2016 Website is down again, apparently there's the need for a domain renewal. Do you guys need any help with that?

**Comment by gordielachance** [comment]
WAW !
Forgot it.
I just paid.
Thanks...

2016-01-22 17:29 GMT+01:00 Morr <notifications@github.com>:

**Comment by Jose-Moreno** [comment]
@gordielachance No need to thank! I think in the future once we grow more as a community we'll have to see if people will be willing to contribute a minimum donation to help you out with the website expenses. I know it's not super expensive, but still I feel later on we should find a way to help you. :+1:

**Comment by gordielachance** [comment]
That is very nice to ask.
If you want, you can make a small donation [here](http://bit.ly/gbreant to) to help paying the host.
But only if you want.  I don't want to receive donations for other's people work (chchwy, feeef, ... are doing far more than me)

**Comment by Jose-Moreno** [comment]
@gordielachance Sorry to bother you man, but today -march 17th 2016- the website is down again. I think we should openly ask for donations at least to maintain the website on a yearly basis. 

Let me know how much is the hosting and the domain name / address costing, so maybe we can get kind of a "tip jar" kind of thing to get that covered. 

I would think it's better paying for the hosting yearly, right? Which is why I'm wondering if something else happened today.

**Comment by gordielachance** [comment]
yes, I just got a mail that says they put the website down because it takes
too much ressources.
My host suggests me to upgrade my plan, but it's much expensive...

I'll try to find a solution soon and to check which price range we are
talking about.

Thanks

2016-03-17 18:18 GMT+01:00 Morr <notifications@github.com>:

**Comment by chchwy** [comment]
I have an idea, maybe we could host the project website on Github pages, it's free.
https://pages.github.com/
Github pages is enough for most of people who come to pencil2d just looking around, downloading executable, or seeing documents.

Only those people which really want to join discussion, they will be redirect to the forum on current pencil2d.org site. I guess it could lower the bandwidth consumption.

What do you think, @gordielachance  ?

**Comment by feeef** [comment]
I like this idea @chchwy !

**Comment by Jose-Moreno** [comment]
@chchwy @gordielachance I think this is the most sensible idea to be honest. 

We can still re-direct most of the word press content pages here using the original pencil2d.org addresses (i.e www.pencil2d.org/faq would take us to the faq page ehre at github).

That way we can leave the forum / community over at the .org wordpress so we can really focus on the community aspect. We can also use this moment to change the theme as we have discussed in the past.

If possible Matt can you give me and gordie mod powers here so I can help to modify the site project?

We are kind of fighting against the clock here, and since it was revealed today that Toonz will be going open source we need to hold this project together.

**Comment by feeef** [comment]
@Jose-Moreno don't worry, Pencil2D will keep living because there are people to make it live! ;)

I have heard about Toonz becoming open source but as a free software supporter, open source doesn't really mean anything to me before seeing what license it will be released on.

Anyway, even if there are other great FOSS for traditional animation, it means that there will just be more choice for the user, which is great! The world of free software is not made of competition. It is made of difference end choice. Pencil2D is so intuitive and we are working on making it lightweight so it will attract many users whatever happens.

BTW, sorry, I haven't been much active with the code lately but hopefully, I will have more time to be back to the mypaint implementation soon!

**Comment by Jose-Moreno** [comment]
@feeef I completely agree. I think one of the strongest points Pencil2D has in it's favor is being easy to use / intuitive, lightweight and also that it is developed by animators (from hobbyist to pros) for animators.

Don't worry take your time, we all have been quite busy lately. I too will be less available during the following months. I'll try to continue providing customer support in the social networks, but I think I won't be able to beta-test the program as much as before.

**Comment by chchwy** [comment]
The Github page website is now live!
https://pencil2d.github.io
Still lacking many things, but this is a beginning. 

@gordielachance @Jose-Moreno 
I have invited you to join the website admin group.

**Comment by gordielachance** [comment]
That's cool guys.  I'll try to put the website back online today.  But maybe that github page is a good solution ?  Do we really still need a wordpress website ? What do u think ?
Either I could bring the website back, either I could redirect pencil2d.org to that github page.
I think the only thing we could miss are the forums, which were quite nice...

**Comment by mapreri** [comment]
probably the content of the wordpress website could be somehow moved entirely to github.
you could have pencil2d.org points to the github.io website (so that the github part is hidden) and moving the forum to a forum.pencil2d.org keeping it where it is now.

**Comment by chchwy** [comment]
Github Pages is purely static (no database behind), so we cannot build a "forum" on github page website. As I said before, I built the github website is to share the excessive bandwidth, I think it can't replace the original wordpress website.

**Comment by gordielachance** [comment]
Ok, i'm currently trying to see with my host if we can put the website back
online...
But they try to sell me a 540€ / year plan (instead of my current 72€/ y);
which I can't afford.
I'm trying to find a way to lower the ressources usage.

2016-03-20 14:14 GMT+01:00 Matt <notifications@github.com>:

**Comment by mapreri** [comment]
@chchwy I think you misunderstood me, I intended to suggest setting up the DNS like:
```
@ IN CNAME pencil2d.github.io.
forum IN A 199.16.128.53
```
(where the IP is where pencil2d.org points now, I don't know if that was where it used to point), the wordpress posts can be moved to jekill, and you can manage the comments with disqus or something similar (and you can import the ones from wordpress)
The forum that needs to be completely dynamic needs to be keep on a separated platform, and the separate domain name assure it (a redirect from pencil2d.org/forum/ → forum.pencil2d.org/ could be set up in jekyll).

This should low the resources needed by a good mean.

**Comment by gordielachance** [comment]
It is back online.
I've enabled new plugins to try to reduce CPU usage.
If it's enough, it will stay online (I hope)
If it is not, they will probably re-disable it tomorrow.

Benoît

2016-03-20 14:25 GMT+01:00 Mattia Rizzolo <notifications@github.com>:

**Comment by Jose-Moreno** [comment]
@gordielachance Thanks a lot for your hard work, right now I'm looking into installing Jekyll to see how I can modify the static page, so we can take some load off the website.

The idea is to maintain the community forums working while everything else is migrated to the Github pages. So we can have tutorials, faq, documentation and everything else here. 

I'm sure we will find a way to make it work! :)

**Comment by Jose-Moreno** [comment]
@chchwy @gordielachance UPDATE: After spending all sunday attempting to install and use Jekyll along with "atom" on my computer to edit the website locally without sucess, I found prose.io which is integrated with Github so we can edit the Github Page through their online editor, even if it's not the best, it does work. However I'm worried that I can't completely change the site layout visually, everything has to be done via code.

I'll continue to work on it when I can, but hopefully we can find a proper visual tool to update the website because I don't have time to code the new page elements from scratch. I'm not a web programmer, so it's getting a bit too complicated for me to code through these markdown and liquid languages to get stuff working.

I did my first commit on the page repo as a test, but it seems I'll have to wait over for Matt to get it approved, which to be honest it isn't ideal, particularly if we're trying to get immediate feedback while designing. For now I'll try to make jekyll work and migrate locally all the other pages from the wordpress site over to to this site layout, before doing a commit. 

Hopefully this will work out, but a great deal of assistance is required. So if anyone has a free day sometime soon it would be awesome so we can understand how this works.

**Comment by gordielachance** [comment]
Ok guys,
The website is down again.
They say there is probably a big problem or a virus that would use the website to make ddos attacks.
I'll put a redirection page to the github project.
Meanwhile, I guess I should find another host for pencil2d; but i'm not sure i'll have time for this for the moment...

**Comment by chchwy** [comment]
Sorry @Jose-Moreno, I didn't set the correct permissions, please try again, I've give you full permission to edit anything. You could edit page directly on Github via pressing the "edit" button at the top-right corner. Prose.io is a good choice, too.

Unfortunately Github pages is designed for web programmer. In my opinion, at beginning the easiest way to build the website is tell me what you want  ex. "I need a FAQ page", I create an empty page, then you fill it. Then you could follow the steps to create 2nd, 3rd pages. Don't worry about asking me anythings :smile:

**Comment by Jose-Moreno** [comment]
@chchwy It's ok It's not like I can't program the page :stuck_out_tongue:  I just wanted to have it running as fast as possible since I have limited time, but don't worry! :smiley: I've already tested the permissions and I could commit directly from my local copy.

I'll try to  update more stuff before committing since I managed to setup the Jekyll viewer. If there's something I just can't do I'll let you know. Thanks a lot for sticking out with Pencil and for helping us as much as you have!

I hope we can continue to nurture this project so it will finally become an important tool for artists out there. I'll keep in touch, cheers :)
